Parse/src/main/java/com/parse/NetworkQueryController.java

```diff
@@ -134,9 +134,12 @@ public Integer then(Task<JSONObject> task) throws Exception {
       if (resultClassName == null) {
         resultClassName = state.className();
       }
+      boolean isSubset = state.selectedKeys() != null;
+      JSONArray selectedKeys = isSubset ? new JSONArray(state.selectedKeys()) : null;
       for (int i = 0; i < results.length(); ++i) {
         JSONObject data = results.getJSONObject(i);
-        T object = ParseObject.fromJSON(data, resultClassName, state.selectedKeys() == null);
+        if (isSubset) data.put(ParseObject.KEY_SELECTED_KEYS, selectedKeys);
```

Comment:
Can you pass in selectedKeys into a method instead that accepts selectedKeys() instead?

https://github.com/ParsePlatform/Parse-SDK-iOS-OSX/blob/master/Parse/PFObject.m#L772-L774

It seems weird that the NetworkQueryController has access to this private key..